Src/WitsmlExplorer.Api/Services/LogDataReader.cs

```diff
@@ -166,14 +166,23 @@ public async Task<WitsmlLogData> GetNextBatch()
 
             try
             {
-                return await _buffer.ReceiveAsync(_receiveTokenSource.Token);
+                if (cancellationToken == null)
+                {
+                    return await _buffer.ReceiveAsync(_receiveTokenSource.Token);
+                }
+                using var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(_receiveTokenSource.Token, cancellationToken.Value);
+                return await _buffer.ReceiveAsync(linkedCts.Token);
             }
-            catch (TaskCanceledException)
+            catch (TaskCanceledException ex)
             {
                 if (_exception != null)
                 {
                     throw _exception;
                 }
+                if (cancellationToken != null && cancellationToken.Value.IsCancellationRequested)
```

Comment:
Can't this also be simplified to `cancellationToken?.ThrowIfCancellationRequested();`?

Reply:
done